In [99]:
import numpy as np

In [100]:
url = 'https://archive.ics.uci.edu/ml/machine-learning-databases/iris/iris.data'
iris = np.genfromtxt(url, delimiter=',', dtype='str')

## 1. iris의 차원, 모양, 크기를 프린트하세요

In [101]:
print(iris.ndim)    # 차원
print(iris.shape)   # 모양
print(iris.size)    # 크기

2
(150, 5)
750


## 2. iris의 2번째 axis에서 5번째 값들을 없애고 iris를 프린트하세요.

In [102]:
iris = iris[:,:4]
iris

array([['5.1', '3.5', '1.4', '0.2'],
       ['4.9', '3.0', '1.4', '0.2'],
       ['4.7', '3.2', '1.3', '0.2'],
       ['4.6', '3.1', '1.5', '0.2'],
       ['5.0', '3.6', '1.4', '0.2'],
       ['5.4', '3.9', '1.7', '0.4'],
       ['4.6', '3.4', '1.4', '0.3'],
       ['5.0', '3.4', '1.5', '0.2'],
       ['4.4', '2.9', '1.4', '0.2'],
       ['4.9', '3.1', '1.5', '0.1'],
       ['5.4', '3.7', '1.5', '0.2'],
       ['4.8', '3.4', '1.6', '0.2'],
       ['4.8', '3.0', '1.4', '0.1'],
       ['4.3', '3.0', '1.1', '0.1'],
       ['5.8', '4.0', '1.2', '0.2'],
       ['5.7', '4.4', '1.5', '0.4'],
       ['5.4', '3.9', '1.3', '0.4'],
       ['5.1', '3.5', '1.4', '0.3'],
       ['5.7', '3.8', '1.7', '0.3'],
       ['5.1', '3.8', '1.5', '0.3'],
       ['5.4', '3.4', '1.7', '0.2'],
       ['5.1', '3.7', '1.5', '0.4'],
       ['4.6', '3.6', '1.0', '0.2'],
       ['5.1', '3.3', '1.7', '0.5'],
       ['4.8', '3.4', '1.9', '0.2'],
       ['5.0', '3.0', '1.6', '0.2'],
       ['5.0', '3.4', '1.6', '0.4'],
 

## 3. iris의 데이터 타입을 float64로 바꾸고, iris의 데이터타입을 프린트하세요

In [103]:
iris = iris.astype(np.float64)
iris.dtype

dtype('float64')

## 4. iris에서 1보다 작은 원소들은 0으로 바꾸고, iris를 프린트하세요.

In [104]:
iris[iris<1] = 0
iris

array([[5.1, 3.5, 1.4, 0. ],
       [4.9, 3. , 1.4, 0. ],
       [4.7, 3.2, 1.3, 0. ],
       [4.6, 3.1, 1.5, 0. ],
       [5. , 3.6, 1.4, 0. ],
       [5.4, 3.9, 1.7, 0. ],
       [4.6, 3.4, 1.4, 0. ],
       [5. , 3.4, 1.5, 0. ],
       [4.4, 2.9, 1.4, 0. ],
       [4.9, 3.1, 1.5, 0. ],
       [5.4, 3.7, 1.5, 0. ],
       [4.8, 3.4, 1.6, 0. ],
       [4.8, 3. , 1.4, 0. ],
       [4.3, 3. , 1.1, 0. ],
       [5.8, 4. , 1.2, 0. ],
       [5.7, 4.4, 1.5, 0. ],
       [5.4, 3.9, 1.3, 0. ],
       [5.1, 3.5, 1.4, 0. ],
       [5.7, 3.8, 1.7, 0. ],
       [5.1, 3.8, 1.5, 0. ],
       [5.4, 3.4, 1.7, 0. ],
       [5.1, 3.7, 1.5, 0. ],
       [4.6, 3.6, 1. , 0. ],
       [5.1, 3.3, 1.7, 0. ],
       [4.8, 3.4, 1.9, 0. ],
       [5. , 3. , 1.6, 0. ],
       [5. , 3.4, 1.6, 0. ],
       [5.2, 3.5, 1.5, 0. ],
       [5.2, 3.4, 1.4, 0. ],
       [4.7, 3.2, 1.6, 0. ],
       [4.8, 3.1, 1.6, 0. ],
       [5.4, 3.4, 1.5, 0. ],
       [5.2, 4.1, 1.5, 0. ],
       [5.5, 4.2, 1.4, 0. ],
       [4.9, 3

## 5. iris와 같은 크기이면서 100번 seed를 이용하여 각각의 값이 standard normal 분포를 따르는 iris2 배열을 만드세요 (shape 속성을 활용하세요), iris2를 프린트하세요.

In [105]:
rng   = np.random.default_rng(seed=100)
iris2 = rng.standard_normal(iris.shape)
iris2

array([[-1.15754965e+00,  2.89755802e-01,  7.80854069e-01,
         5.43973645e-01],
       [-9.61382641e-01,  1.07100867e+00,  7.01455660e-01,
         7.04973455e-01],
       [ 7.45062603e-01,  1.10434724e+00,  2.24297240e+00,
        -6.11493123e-01],
       [ 4.72111831e-02,  1.75423468e+00, -1.33797987e+00,
         3.25574469e-01],
       [-6.89117716e-01, -1.98218100e-02,  4.74753246e-01,
        -1.93110142e+00],
       [-9.92478278e-01, -1.40547108e+00, -2.31095508e-01,
        -6.88847085e-01],
       [ 1.51510578e+00, -6.03171547e-01,  1.71368449e+00,
        -4.06249191e-01],
       [ 2.71409521e-01,  3.98402001e-02,  1.15183183e-02,
        -1.12717801e+00],
       [ 3.34712975e-01,  3.83891549e-01,  2.37835514e-01,
         6.21411674e-01],
       [-8.19245547e-01, -2.97579206e-01, -6.61562705e-01,
        -1.70417033e+00],
       [ 3.67536321e-01, -6.35488793e-01, -7.77205069e-02,
         2.24976784e+00],
       [ 2.30310529e-01,  1.07415900e-01,  1.07394578e+00,
      

## 6. iris2의 소수부를 버림하고 프린트하세요.

In [106]:
remainder, whole_part = np.modf(iris2)
iris2 = whole_part
iris2

array([[-1.,  0.,  0.,  0.],
       [-0.,  1.,  0.,  0.],
       [ 0.,  1.,  2., -0.],
       [ 0.,  1., -1.,  0.],
       [-0., -0.,  0., -1.],
       [-0., -1., -0., -0.],
       [ 1., -0.,  1., -0.],
       [ 0.,  0.,  0., -1.],
       [ 0.,  0.,  0.,  0.],
       [-0., -0., -0., -1.],
       [ 0., -0., -0.,  2.],
       [ 0.,  0.,  1.,  1.],
       [ 1., -0.,  1., -0.],
       [-1., -0.,  1.,  0.],
       [ 1., -0., -0., -1.],
       [-1., -0.,  1.,  1.],
       [-0.,  1., -0.,  0.],
       [ 0., -0., -0.,  0.],
       [-0., -0.,  0., -0.],
       [ 0., -0.,  0., -0.],
       [-0., -1.,  1.,  0.],
       [ 0., -1.,  0., -0.],
       [ 1., -1.,  0.,  1.],
       [-1.,  0., -1.,  0.],
       [ 0.,  1.,  0., -0.],
       [ 0.,  0.,  1., -0.],
       [ 1.,  0.,  0.,  1.],
       [-0., -0.,  0., -0.],
       [ 0., -0., -0.,  0.],
       [ 1., -1.,  1.,  0.],
       [ 1.,  0., -0., -0.],
       [-0., -1., -0.,  0.],
       [-0.,  0., -1.,  1.],
       [-0.,  0.,  0., -0.],
       [ 2.,  

## 7. 배열의 각 위치에서 iris2의 절댓값이 1보다 크면, iris의 값을 아니면 0의 값을 갖는 배열 iris3를 만들고 프린트하세요.

In [107]:
iris3 = np.where(np.abs(iris2)>1, iris, 0)
iris3

array([[0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 1.3, 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [0. , 0. , 0. , 0. ],
       [4.9, 0

In [93]:
import pandas as pd
from pandas import Series, DataFrame
url = 'https://archive.ics.uci.edu/ml/machine-learning-databases/iris/iris.data'
iris = np.genfromtxt(url, delimiter=',', dtype='str')

## 8. column을 순서대로 a,b,c,d,e 로 갖는 데이터프레임의 형태로 iris를 만들고 프린트하세요. 

In [94]:
iris = DataFrame(iris, columns=['a', 'b', 'c', 'd', 'e'])
iris

a    b    c    d               e
0    5.1  3.5  1.4  0.2     Iris-setosa
1    4.9  3.0  1.4  0.2     Iris-setosa
2    4.7  3.2  1.3  0.2     Iris-setosa
3    4.6  3.1  1.5  0.2     Iris-setosa
4    5.0  3.6  1.4  0.2     Iris-setosa
..   ...  ...  ...  ...             ...
145  6.7  3.0  5.2  2.3  Iris-virginica
146  6.3  2.5  5.0  1.9  Iris-virginica
147  6.5  3.0  5.2  2.0  Iris-virginica
148  6.2  3.4  5.4  2.3  Iris-virginica
149  5.9  3.0  5.1  1.8  Iris-virginica

[150 rows x 5 columns]

## 9.   iris의 column a,b,c,d에 대해서 데이터 타입을 float64로 바꾸고 프린트하세요.

In [96]:
iris[['a','b','c','d']] = iris[['a','b','c','d']].astype(np.float64)
print(iris[['a','b','c','d']].dtypes)
iris

a    float64
b    float64
c    float64
d    float64
dtype: object


a    b    c    d               e
0    5.1  3.5  1.4  0.2     Iris-setosa
1    4.9  3.0  1.4  0.2     Iris-setosa
2    4.7  3.2  1.3  0.2     Iris-setosa
3    4.6  3.1  1.5  0.2     Iris-setosa
4    5.0  3.6  1.4  0.2     Iris-setosa
..   ...  ...  ...  ...             ...
145  6.7  3.0  5.2  2.3  Iris-virginica
146  6.3  2.5  5.0  1.9  Iris-virginica
147  6.5  3.0  5.2  2.0  Iris-virginica
148  6.2  3.4  5.4  2.3  Iris-virginica
149  5.9  3.0  5.1  1.8  Iris-virginica

[150 rows x 5 columns]

## 10. iris의 column과 index 순서를 거꾸로 갖는 iris2를 만드세요. (e->a, 149->0) iris2를 프린트하세요.

In [97]:
iris2 = iris.sort_index(ascending=False).sort_index(axis='columns',ascending=False)
iris2

e    d    c    b    a
149  Iris-virginica  1.8  5.1  3.0  5.9
148  Iris-virginica  2.3  5.4  3.4  6.2
147  Iris-virginica  2.0  5.2  3.0  6.5
146  Iris-virginica  1.9  5.0  2.5  6.3
145  Iris-virginica  2.3  5.2  3.0  6.7
..              ...  ...  ...  ...  ...
4       Iris-setosa  0.2  1.4  3.6  5.0
3       Iris-setosa  0.2  1.5  3.1  4.6
2       Iris-setosa  0.2  1.3  3.2  4.7
1       Iris-setosa  0.2  1.4  3.0  4.9
0       Iris-setosa  0.2  1.4  3.5  5.1

[150 rows x 5 columns]

## 11. iris2의 column a와 column  b의 correlation을 프린트하세요.

In [98]:
iris2['a'].corr(iris2['b'])

-0.10936924995064937